src/components/Select/Select.component.tsx

```diff
@@ -0,0 +1,191 @@
+        setValue('');
+        if (isOpen) {
+            inputRef.current.blur();
+            setIsOpen(false);
+        } else {
+            inputRef.current.focus();
+            setIsOpen(true);
+        }
+    };
+
+    /**
+	 * On clicking anywhere other than `input field` or `trailing icon` the dropdown closes
+	 */
+    const onDropDownCloseHandler = (e: MouseEvent) => {
+        if (e.target === inputRef.current) {
+            return;
+        }
+        if (e.target === trailingIconRef.current) {
+            return;
+        }
+        if (isOpen) {
+            setIsOpen(false);
+        }
+    };
+
+    /**
+	 * On user selecting the option the value / label of the menu Item is set to the  input field
+	 */
+    const onUserSelectHandler = (
+        e: React.MouseEvent<HTMLLIElement, MouseEvent>,
+        option: OptionType,
+    ) => {
+        setValue(option.label ?? option.value);
+        onSelectOptionHandler(e, option);
+    };
+
+    // On clicking anywhere other than the input field the dropdown closes
+    useEffect(() => {
+        document.body.addEventListener('click', onDropDownCloseHandler);
+
+        return () => {
+            document.body.removeEventListener('click', onDropDownCloseHandler);
+        };
+    });
+
+    return (
+        <>
+            <Wrapper>
+                <Input
+                    ref={inputRef}
+                    placeholder=' '
+                    type='text'
+                    onFocus={onFocusHandler}
+                    readOnly={true}
+                    value={value}
+                    leadingIcon={leadingIcon}
+                />
+                <Label leadingIcon={leadingIcon}>{label}</Label>
+                {leadingIcon && (
+                    <LeadingIcon className='select__leading-icon'>
+                        <Icon
+                            name={leadingIcon}
+                            size={16}
+                        />
+                    </LeadingIcon>
+                )}
+                <TrailingIcon
+                    className='select__trailing-icon'
+                    ref={trailingIconRef}
+                    onClick={onIconTrailingIconClickHandler}
+                >
+                    {isOpen || value ? (
+                        <Icon
+                            name='Close'
+                            size={16}
+                        />
+                    ) : (
+                        <Icon
+                            name='ArrowDown'
+                            size={16}
+                        />
+                    )}
+                </TrailingIcon>
+            </Wrapper>
+            <Options
+                open={isOpen}
+                className='select__option-list'
+            >
+                {options.map((option) => (
+                    <MenuItem
+                        key={option.value}
+                        className={(option.label ?? option.value) === value ? 'active' : ''}
```

Comment:
```suggestion
                        className={(option.label ?? option.value) === value && 'active'}
```

if type is not supported then update the type definition also